src/components/RegistrationComponent/RegistrationComponent.js

```diff
@@ -0,0 +1,116 @@
+import React, { useState } from "react";
+import { useHistory } from "react-router-dom";
+import axios from "axios";
+import Snackbar from "@material-ui/core/Snackbar";
+import logo from "../../icons/Vector.svg";
+import image from "../../icons/image.svg";
+import "./RegistrationComponent.scss";
+
+const RegistrationFormComponent = () => {
+  const history = useHistory();
+  const regExLogin = /(\w+).{6}/;
+  const regExPassword = /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{1,}$/;
+  const [open, setOpen] = useState({ stateSnackbar: false, message: "" });
+  const { stateSnackbar, message } = open;
+
+  const checkPass = "Check password";
+  const checkLogin = "Check login";
+  const checkPasswords = "Check passwords";
+
+  const booleanLoginFunction = () => {
+    setOpen({
+      stateSnackbar: true,
+      message: checkLogin,
+    });
+  };
+
+  const booleanPasswordFunction = () => {
+    setOpen({
+      stateSnackbar: true,
+      message: checkPass,
+    });
+  };
+
+  const checkPasswordFunction = () => {
+    setOpen({
+      stateSnackbar: true,
+      message: checkPasswords,
+    });
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  const registrationPage = () => {
+    history.push("/autorisation");
+  };
+
+  const getDataForms = (e) => {
+    e.preventDefault();
+    const formData = new FormData(e.target);
+    const formLogin = formData.get("login").trim();
+    const formPassword = formData.get("password");
+    const formPasswordRepeat = formData.get("passwordRepeat");
+    if (regExLogin.test(formLogin)) {
+      booleanLoginFunction();
+    }
+    if (formPassword === formPasswordRepeat) {
+      checkPasswordFunction();
+    }
+    if (regExPassword.test(formPassword)) {
```

Comment:
why is the functionality for matching the equality of passwords first, and only then checking for a pattern?